akka-stream/src/main/scala/akka/stream/impl/fusing/GraphInterpreter.scala

```diff
@@ -222,6 +222,38 @@ private[stream] object GraphInterpreter {
       assembly
     }
   }
+
+  /**
+   * INTERNAL API
+   */
+  private val _currentInterpreter = new ThreadLocal[Array[AnyRef]] {
+    /*
+     * Using an Object-array avoids holding on to the GraphInterpreter class
+     * when this accidentally leaks onto threads that are not stopped when this
+     * class should be unloaded.
+     */
+    override def initialValue = new Array(1)
```

Comment:
Sneaky, I get it - we avoid hitting the thread-local, but just touch the pulled out array after it's pulled out... nice.